main/src/ca/uwaterloo/flix/api/Bootstrap.scala

```diff
@@ -91,44 +92,50 @@ object Bootstrap {
     }
   }
 
-}
-
-class Bootstrap {
-
-  // Timestamps at the point the sources were loaded
-  private var timestamps: Map[Path, Long] = Map.empty
-
-  // Lists of paths to the source files, flix packages and .jar files used
-  private var sourcePaths: List[Path] = List.empty
-  private var flixPackagePaths: List[Path] = List.empty
-  private var mavenPackagePaths: List[Path] = List.empty
-
   /**
+    * Creates a new Bootstrap object and initializes it.
     * If a `flix.toml` file exists, parses that to a Manifest and
     * downloads all required files. Otherwise checks the /lib folder
-    * to see what dependencies are already downloadet.
-    * Then returns a list of all flix source files, flix packages
-    * and .jar files that this project uses.
+    * to see what dependencies are already downloadet. Also finds
+    * all .flix source files.
+    * Then returns the initialized Bootstrap object or an error.
     */
-  def bootstrap(pathString: String): Result[List[Path], BootstrapError] = {
+  def bootstrap(path: Path)(implicit out: PrintStream): Result[Bootstrap, BootstrapError] = {
     //
     // Determine the mode: If `path/flix.toml` exists then "project" mode else "folder mode".
     //
-    val path = Paths.get(pathString)
+    val bootstrap = new Bootstrap()
     val tomlPath = Bootstrap.getManifestFile(path)
-    if (Files.exists(tomlPath)) {
-      projectMode(path)
+    (if (Files.exists(tomlPath)) {
+      out.println("Found flix.toml")
```

Comment:
-> Found 'flix.toml". Entering project mode. Will now check dependencies. 

-> Did not find 'flix.toml'. Entering folder mode. No dependencies will be downloaded.